src/main/java/school/hei/haapi/endpoint/rest/controller/StudentFileController.java

```diff
@@ -1,22 +1,59 @@
 package school.hei.haapi.endpoint.rest.controller;
 
+import static java.util.stream.Collectors.toUnmodifiableList;
 import static org.springframework.http.MediaType.APPLICATION_PDF_VALUE;
+import static org.springframework.http.MediaType.MULTIPART_FORM_DATA_VALUE;
 
+import java.util.List;
 import lombok.AllArgsConstructor;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestPart;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import school.hei.haapi.endpoint.rest.mapper.FileInfoMapper;
+import school.hei.haapi.endpoint.rest.model.FileInfo;
+import school.hei.haapi.endpoint.rest.model.FileType;
 import school.hei.haapi.service.StudentFileService;
 
 @RestController
 @AllArgsConstructor
 public class StudentFileController {
   private final StudentFileService fileService;
+  private final FileInfoMapper fileInfoMapper;
 
   @GetMapping(
       value = "/students/{id}/scholarship_certificate/raw",
       produces = APPLICATION_PDF_VALUE)
   public byte[] getStudentScholarshipCertificate(@PathVariable(name = "id") String studentId) {
     return fileService.generatePdf(studentId, "scolarity");
   }
+
+  @PostMapping(value = "/students/{student_id}/files/raw", consumes = MULTIPART_FORM_DATA_VALUE)
+  public FileInfo uploadStudentFile(
+      @PathVariable(name = "student_id") String studentId,
+      @RequestParam(name = "file_name") String fileName,
```

Comment:
@RequestParam("filename") is not needed anymore since multipart conserves the original filename